tests/unit/no-top-level-side-effects.test.ts

```diff
@@ -3013,19 +3311,6 @@ const invalid: RuleTester.InvalidTestCase[] = [
           endColumn: 19
         }
       ]
-    },
-    {
-      code: `const x = { foo: (function(){ return 1 })() };`,
-      options: [options.allowIIFE],
-      errors: [
-        {
-          messageId: '0',
-          line: 1,
-          column: 18,
-          endLine: 1,
-          endColumn: 44
-        }
-      ]
```

Comment:
This test has been moved to the `// Immediately Invoked Functions Expressions (IIFE), when allowed` group.